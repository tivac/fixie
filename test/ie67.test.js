"use strict";

var simple = require("./util.js").simple,
    nested = require("./util.js").nested;

describe(":ie67", () => {
    it("should transform :ie67(...) into an ie67 hack (single selector)", () =>
        simple(`
            :ie67(.fooga) {
                color: red;
            }
        `)
    );

    it("should transform :ie67(...) into an ie67 hack (multiple selectors)", () =>
        simple(`
            :ie67(.fooga .booga) {
                color: red;
            }
        `)
    );

    it("should transform :ie67(...) into an ie67 hack (multiple declarations)", () =>
        simple(`
            :ie67(.fooga) { color: red; background: blue; }
        `)
    );

    it("should work with postcss-nested", () =>
        nested(`
            .fooga {
                color: red;
                
                :ie67(&) {
                    color: blue;
                }
            }
        `)
    );
});
