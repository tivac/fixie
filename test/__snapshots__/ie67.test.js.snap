// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`:ie67 should transform :ie67(...) into an ie67 hack (multiple declarations) 1`] = `".fooga { *color: red; *background: blue; }"`;

exports[`:ie67 should transform :ie67(...) into an ie67 hack (multiple selectors) 1`] = `
".fooga .booga {
    *color: red;
}"
`;

exports[`:ie67 should transform :ie67(...) into an ie67 hack (single selector) 1`] = `
".fooga {
    *color: red;
}"
`;

exports[`:ie67 should work with postcss-nested 1`] = `
".fooga {
    color: red;
}
    
    .fooga {
        *color: blue;
    }"
`;
